pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/CassandraSinkTask.java

```diff
@@ -189,28 +199,22 @@ public void write(Record<T> record) throws Exception {
         log.info("raw {}", rawvalue);
       }
     }
-    int number;
-    synchronized (this) {
-      incomingList.add(record);
-      number = incomingList.size();
-    }
-    if (number == batchSize) {
+
+    incomingQueue.put(record); // blocking call to slow down the topic consumer
```

Comment:
this is not working
because at this point the actTimeout already started on the message.